pkg/endpoint/policy.go

```diff
@@ -552,6 +553,7 @@ func (e *Endpoint) SetIdentity(identity *identityPkg.Identity) {
 		oldIdentity = e.SecurityIdentity.StringID()
 	}
 
+	identitymanager.Upsert(identity)
```

Comment:
This is fine, I think we should do an `identitymanager.Remove(oldIdentity)` here as well.

Reply:
LMK what you think of the new behavior.